### PR TITLE
Fix Linux support on OBV

### PR DESCRIPTION
**Pull request by reuben**
Extension loads fine, but there are still a few things to sort out:
- Loading any modules not registered by Viper crashes the server (except built-in modules like `os` and `sys`).
- You must link libpython2.5.so.1.0 from `lib/plat-linux2` to `orangebox/bin` for the extension to load.

### CODE REVIEW

**Comment by theY4Kman** [comment]
1. See diff comments on extension.cpp lines 171 and 172
2. See diff comments on viper_globals.h on line 154

Just wanted to post this for the record.

File: src/viper_globals.h [hunk 1]
```diff
@@ -100,7 +100,7 @@ extern PyObject *g_pViperException;
     if (pPlugin == NULL) \
     { \
         PyErr_SetString(g_pViperException, "Error retrieving current plug-in"); \
-        return NULL; \
+        return 0; \
```
Comment (theY4Kman) [inline src/viper_globals.h:103]: There are a few places where GET_THREAD_PLUGIN is used in a function where a pointer is expected (such as usermessages__hook in py_usermessages.cpp). The 0 would, presumably, provide the same incorrect type errors that returning NULL in a function that expects an int would. Is this true? If so, a better solution to GET_THREAD_PLUGIN would be needed. If you agree and this is true, I'll try to come up with something.

File: src/extension.cpp [hunk 2]
```diff
@@ -168,33 +169,23 @@ ViperExtension::SDK_OnLoad(char *error, size_t maxlength, bool late)
     }
 #else
     /* We must load in the binary to allow access to it.
-     * Thanks to your-name-here for that bit of info!
-     */
+    * Thanks to your-name-here for that bit of info!
```
Comment (theY4Kman) [inline src/extension.cpp]: I put the spaces in to keep the asterisks and the paragraph in the same alignment. Why did you remove the spaces here?

File: src/extension.cpp [hunk 3]
```diff
@@ -168,33 +169,23 @@ ViperExtension::SDK_OnLoad(char *error, size_t maxlength, bool late)
...
-        strncpy(error, "Could not find SendProxy_EHandleToInt: entity property type "
-                "autodetection would fail without it. Unloading.", maxlength);
-        return false;
-    }
-    
-    void *handle = dlopen(info.dli_fname, RTLD_NOW);
-    if (handle == NULL)
+
+    SourceMod::sm_sendprop_info_t propinfo;
+    bool ret = gamehelpers->FindSendPropInfo("CBasePlayer", "m_hOwnerEntity", &propinfo);
```
Comment (theY4Kman) [inline src/extension.cpp]: psychonic is a genius. This makes so more sense. I wanted to be completely sure that Viper worked in every mod, even if they didn't support m_hOwnerEntity, but mods don't really have a choice, right? The engine imposes m_hOwnerEntity (I assume?).

But I suppose I contradicted my opinions, because I hardcoded SendProxy_EHandleToInt :P (Because, hey, it worked on all the mods I tested it on ;D) Mods didn't really have to support SendProxy_EHandleToInt, though.

File: src/extension.cpp (hunk 2 above)
Comment (reuben) [inline src/extension.cpp]: That was my editor, I didn't notice it in the commit before I sent it. I'll attach a commit to the pull request with the fix.

File: src/extension.cpp (hunk 3 above)
Comment (reuben) [inline src/extension.cpp]: Indeed he is :)

File: src/viper_globals.h (hunk 1 above)
Comment (reuben) [inline src/viper_globals.h:103]: GCC only complains about implicit conversion from `NULL` to `int`, not `int` to `NULL`.
And NULL is just a define for 0, there isn't really any real NULL in C++, so types won't break just because of that.

Comment (theY4Kman) [inline src/viper_globals.h:103]: I was just assuming there might be a problem because NULL is a pointer to 0, rather than an integer, but if implicit conversion works without complaint, I'm game. It compiles without warning on Windows.

File: src/python/py_menus.cpp [hunk 4]
```diff
@@ -154,8 +154,8 @@ public:
     
     ViperMenu(PyObject *myobj, IViperPluginFunction *title_func,
         IViperPluginFunction *callback, IViperPluginFunction *options)
-        : m_Callback (callback), m_TitleFunc(title_func), m_Title(NULL),
-          m_OptionsFunction(options), m_Options(NULL), m_MyObject(myobj)
+        : m_Options(NULL), m_Title(NULL), m_TitleFunc(title_func),
```
Comment (theY4Kman) [inline src/python/py_menus.cpp:157]: wat :P

Comment (reuben) [inline src/python/py_menus.cpp:157]: You're supposed to initialize them at the same order they've been declared :P
I think it only complains with -Wall enabled, though.

Comment (theY4Kman) [inline src/python/py_menus.cpp:157]: Ho damn, I never knew. Awesome, I'll remember that!
